docs/how-to-guides/develop-a-tool/create-your-own-custom-strong-type-connection.md

```diff
@@ -0,0 +1,107 @@
+
+For other connections types, please refer to [Connections](https://microsoft.github.io/promptflow/concepts/concept-connections.html).
+
+## Prerequisites
+- Please ensure that your [Prompt flow for VS Code](https://marketplace.visualstudio.com/items?itemName=prompt-flow.prompt-flow) is updated to at least version 1.2.1.
+- Please install promptflow package and ensure that its version is 0.1.0b8 or later.
+  ```
+  pip install promptflow>=0.1.0b8
+  ```
+
+## Create your own custom strong type connection
+Take [this file](https://github.com/microsoft/promptflow/blob/main/examples/tools/tool-package-quickstart/my_tool_package/tools/tool_with_custom_strong_type_connection.py) as an example, you can create your custom strong type connection as belows:
+
+```python
+from promptflow.connections import CustomStrongTypeConnection
+from promptflow.contracts.types import Secret
+
+
+class MyCustomConnection(CustomStrongTypeConnection):
+    """My custom strong type connection.
+
+    :param api_key: The api key.
+    :type api_key: Secret
+    :param api_base: The api base.
+    :type api_base: String
+    """
+    api_key: Secret
+    api_base: str = "This is a fake api base."
+
+```
+
+Make sure that you adhere to the following guidelines:
+
+* You can define your own custom connection using any desired name, but make sure it inherits from the `CustomStrongTypeConnection` class.
+  > [!Note] Please avoid using the `CustomStrongTypeConnection` class directly.
+* Use the `Secret` type to denote that a key should be treated as a secret. This enhances security by scrubbing secret keys.
+* You have the option to either include your custom connection class within your custom tool or separate it into a distinct Python file.
+* Improve clarity for users by documenting your custom strong type connection using docstrings. Use `param` and `type` to provide explanations for each key, as shown in the following example:
+  
+  ```python
+  """My custom strong type connection.
+
+  :param api_key: The api key get from "https://xxx.com".
+  :type api_key: Secret
+  :param api_base: The api base.
+  :type api_base: String
+  """
+  ```
+  
+  ```yaml
+  $schema: https://azuremlschemas.azureedge.net/promptflow/latest/CustomStrongTypeConnection.schema.json
+  name: "to_replace_with_connection_name"
+  type: custom
+  custom_type: MyCustomConnection
+  module: my_tool_package.tools.my_tool_with_custom_strong_type_connection
+  package: test-custom-tools
+  package_version: 0.0.2
+  configs:
+    api_base: "This is a fake api base." # String type. The api base.
+  secrets:
+    api_key: <user-input> #  Secret type. The api key get from "https://xxx.com". Don't replace the '<user-input>' placeholder. The application will prompt you to enter a value when it runs.
+  ```
+
+## Develop a flow using package tools with custom strong type connections
```

Comment:
Once you've created a custom connection, use it in your flows:

**With Package Tools:**

1. Install your tool package containing the connection.

2. Create a new connection or select "Add Connection" in a tool node. 

3. Fill in the connection keys based on your defined types.

4. Run the flow.

**With Script Tools:** 

1. Create a `CustomConnection` and fill in keys.

2. Reference the connection in your script.

3. Run the flow.